tests/components/matter/test_update.py

```diff
@@ -169,3 +194,185 @@ async def test_update_install(
     state = hass.states.get("update.mock_dimmable_light")
     assert state.state == STATE_OFF
     assert state.attributes.get("installed_version") == "v2.0"
+
+
+async def test_update_install_failure(
+    hass: HomeAssistant,
+    matter_client: MagicMock,
+    check_node_update: AsyncMock,
+    update_node: AsyncMock,
+    updateable_node: MatterNode,
+) -> None:
+    """Test update entity service call errors."""
+    state = hass.states.get("update.mock_dimmable_light")
+    assert state
+    assert state.state == STATE_OFF
+    assert state.attributes.get("installed_version") == "v1.0"
+
+    await async_setup_component(hass, "homeassistant", {})
+
+    check_node_update.return_value = MatterSoftwareVersion(
+        vid=65521,
+        pid=32768,
+        software_version=2,
+        software_version_string="v2.0",
+        firmware_information="",
+        min_applicable_software_version=0,
+        max_applicable_software_version=1,
+        release_notes_url="http://home-assistant.io/non-existing-product",
+        update_source=UpdateSource.LOCAL,
+    )
+
+    await hass.services.async_call(
+        "homeassistant",
+        "update_entity",
+        {
+            ATTR_ENTITY_ID: "update.mock_dimmable_light",
+        },
+        blocking=True,
+    )
+
+    assert matter_client.check_node_update.call_count == 2
+
+    state = hass.states.get("update.mock_dimmable_light")
+    assert state
+    assert state.state == STATE_ON
+    assert state.attributes.get("latest_version") == "v2.0"
+    assert (
+        state.attributes.get("release_url")
+        == "http://home-assistant.io/non-existing-product"
+    )
+
+    update_node.side_effect = UpdateCheckError("Error finding applicable update")
+
+    with pytest.raises(HomeAssistantError):
+        await hass.services.async_call(
+            "update",
+            "install",
+            {
+                ATTR_ENTITY_ID: "update.mock_dimmable_light",
+                "version": "v3.0",
+            },
+            blocking=True,
+        )
+
+    update_node.side_effect = UpdateError("Error updating node")
+
+    with pytest.raises(HomeAssistantError):
+        await hass.services.async_call(
+            "update",
+            "install",
+            {
+                ATTR_ENTITY_ID: "update.mock_dimmable_light",
+                "version": "v3.0",
```

Comment:
Also please use constants here

Reply:
Is using constants really a good idea in all cases? E.g. when using constants in a test, a change of that constant might actually break the user facing API (e.g. in this case users might have `version` hard coded in an automation or something). So changing the constant should break this test. So a hard coded value is better, IMHO.

Reply:
Hm, I guess that is the API of the update component, so a breaking change would not really be a concern of this particular implementation (on the Matter end).

I've replaced things with constants :+1: